pkg/kubernetes/artifact.go

```diff
@@ -0,0 +1,70 @@
+package kubernetes
+
+import (
+	"strings"
+
+	clouddriver "github.com/homedepot/go-clouddriver/pkg"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// ReplaceDockerImageArtifacts finds containers in a given unstructured
+// Kubernetes object and replaces the `image` value of the container
+// with the artifact reference if the artifact name matches the
+// container's `image` value.
+//
+// Pods define containers at the JSON path ".spec.containers",
+// all other kinds defines containers at the JSON path ".spec.template.spec.containers".
+func ReplaceDockerImageArtifacts(u *unstructured.Unstructured,
+	artifacts map[string]clouddriver.TaskCreatedArtifact) error {
+	var (
+		containers []interface{}
+		found      bool
+		err        error
+	)
+
+	if len(artifacts) == 0 {
+		return nil
+	}
+
+	if strings.EqualFold(u.GetKind(), "pod") {
+		containers, found, err = unstructured.NestedSlice(u.Object, "spec", "containers")
+	} else {
+		containers, found, err = unstructured.NestedSlice(u.Object, "spec", "template", "spec", "containers")
```

Comment:
I failed to give full requirements in the story, but will need to get `initContainers` as well

See the [clouddriver code](https://github.com/spinnaker/clouddriver/blob/58ab154b0ec0d62772201b5b319af349498a4e3f/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/artifact/Replacer.java#L166)

Reply:
Arg! I missed that.